Java/Operating-System/alloc.java

```diff
@@ -0,0 +1,172 @@
+/* 
+Title: Memory Allocation in Operating Systems in Java
+
+Description:
+(a) First-Fit: Allocate the first hole that is big enough. Searching can start either at the beginning of the set of holes or where the previous first search ended. We can stop searching as soon as we find a free hole that is large enough.
+(b) Best-Fit: Allocate the smallest hole that is big enough. We must search the entire list unless the list is ordered by size.
+(c) Worst-Fit: Allocate the largest hole. We must search the entire list unless the list is ordered by size.
+ 
+*/
+
+import java.util.ArrayList;
+import java.util.Scanner;
+public class alloc {
+
+  public static void main(String[] args) {
```

Comment:
Keep your main() function only for the Driver Code. Break down your entire code into modular functions and then call them from the main() function